src/nvim/eval/funcs.c

```diff
@@ -9088,7 +9186,7 @@ static void f_setreg(typval_T *argvars, typval_T *rettv, FunPtr fptr)
         yank_type = kMTBlockWise;
         if (ascii_isdigit(stropt[1])) {
           stropt++;
-          block_len = getdigits_long((char_u **)&stropt, true, 0) - 1;
+          block_len = getdigits_long((char_u **)&stropt, false, 0) - 1;
```

Comment:
Are these worth adding a test?

Reply:
if Vim doesn't emsg then it's a "feature" of setreg().